src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActions/FixAllSuggestedAction.cs

```diff
@@ -64,14 +65,15 @@ public override bool TryGetTelemetryId(out Guid telemetryId)
         public string GetDiagnosticID()
             => Diagnostic.GetTelemetryDiagnosticID();
 
-        protected override void InnerInvoke(
+        protected override async Task InnerInvokeAsync(
             IProgressTracker progressTracker, CancellationToken cancellationToken)
         {
+            // Even though we're async, we should still be on the UI thread at this point.
```

Comment:
❔ If the path is asynchronous, why not just switch to the main thread if/when necessary?

Reply:
soryr, i should have explained in the PR.  this is more to get the test pass to run and to see if i'm incorrect about anything here.  you're absolutely right that the real code should do the proper switch.

Reply:
(note: i've made that change in the (part 2) pr).